java/debugger.jpda/src/org/netbeans/modules/debugger/jpda/models/ShortenedStrings.java

```diff
@@ -141,25 +148,59 @@ static String getStringWithLengthControl(StringReference sr) throws InternalExce
                             continue;
                         }
                         Type type = f.type();
-                        if (type instanceof ArrayType &&
-                            "char".equals(((ArrayType) type).componentTypeName())) {
-                            valuesField = f;
+                        if (type instanceof ArrayType) {
+                            String componentType = ((ArrayType)type).componentTypeName();
+                            if ("byte".equals(componentType)){
+                                isCompressedImpl = true;
+                                valuesField = f;
+                            }
+                            else if ("char".equals(componentType)){
+                                valuesField = f;
+                            }
+                            else{
+                                continue;
+                            }
                             break;
                         }
                     }
                 }
+                else if (valuesField.type() instanceof ArrayType &&
+                        "byte".equals(((ArrayType)valuesField.type()).
+                                componentTypeName())){
+                    isCompressedImpl = true;
+                }
                 if (valuesField == null) {
                     isShort = true; // We did not find the values field.
                 } else {
+                    if (isCompressedImpl){
+                        //is it compressed?
+                        final int LATIN1 = 0;
+                        Field coderField = ReferenceTypeWrapper.fieldByName(st,
+                                "coder");
+                        Value coderValue;
+                        if (coderField != null){
+                            coderValue = ObjectReferenceWrapper.getValue(sr,
+                                    coderField);
+                            if (coderValue instanceof IntegerValue &&
+                                    ((IntegerValue)coderValue).value() != LATIN1){
```

Comment:
This check failed on JDK 17 for me. I got a `ByteValue` at this point. Instead of getting to narrow just switch to `PrimitiveValue`, which has helpful accessors.

```suggestion
                            if (coderValue instanceof PrimitiveValue &&
                                    ((PrimitiveValue)coderValue).intValue() != LATIN1){
```